meetings/2021-07/july-15.md

```diff
@@ -0,0 +1,734 @@
+
+GCL: Okay, that's interesting. I guess - yeah, I guess it'll just have to be thought about separately.
+
+CP:  Again, you will go from throwing an error to allowing that object to be passed around, but you have to explore that at some point.
+
+USA: That's it. Surprisingly, there is nothing else on the queue.
+
+LEO: We've had the topics from Daniel from the last meeting that I still want to give space for them to present. But I at this that we were discussing these constraints I wonder if it's time again that we can ask for For the stage advancement, 
+
+USA: I guess you could ask, I guess you could ask again And in after that we could see if anyone would like to talk about it
+
+LEO: So Jordan, are you feeling about it? I know you bring the main objection so far.
+
+JHD: I mean So, I have a few classes of concerns. So one is that this feels like a really weird and Magic API for me like it's very clever solution to the problem for And I hear that, that, it achieves the goals that people who want membranes have because it lets those use a membrane Library, still, but then it gives you better results with it and you know, takes more of the complexity and puts it in the platform. I get back. And hear that amp has a use case for which membranes work, but our Overkill and that this all works in a more granular passion. But the use cases that this proposal Our include those and more. And that's what we went into Stage 2 with to solve all of those problems, regardless of what the original stated motivation was, that was the Assumption, at least that I was under, and I'm assuming that everyone else was under, when it went into stage 2. Then the other classic concerned separate from the like kind of strangeness of this new mechanism. The other concern is that then happens with those use cases. And I've certainly seen some examples like you've shown and heard some arguments that it's easy - if you have direct object access it is easy to get things wrong and that is certainly true. And it is certainly likely that there are alternative approaches than direct object access to another realm that would achieve the goals that I'm concerned about in ways that are harder to get wrong. I'm concerned that if this advances to stage 3 with this new form that has a restricted - like a subset of use cases met, that the sort of implied consensus that we also want to solve those other problems. Even if everyone may not have agreed with that like it would I would say implied is kind of gone. And that means that if I want to try and solve that use case, not only do I have to start from scratch with a new proposal, but I also have to come up with a mechanism that a lot of very experienced and intelligent people have not come up with, and admittedly weren't specifically trying to come up with, but it just kind of it's very strange to me that within stage to the shape of a solution would change so drastically such that it's almost a completely different proposal.
```

Comment:
```suggestion
LEO: So JHD, are you feeling about it? I know you bring the main objection so far.

JHD: I have a few classes of concerns. So one is that this feels like a really weird and magic API for me. It's a very clever solution to the problem and I hear that it achieves the goals that people who want membranes have because it lets them use a membrane library still, but then it gives you better results with it and you know, takes more of the complexity and puts it in the platform. I get that. And I hear that AMP has a use case for which membranes work, but are overkill and that this all works in a more granular way. But this does not solve all of the use cases that this proposal went into Stage 2 with a solution for, regardless of what the original stated motivation was. That was the assumption at least that I was under, and I'm assuming that everyone else was under, when it went into stage 2. Then the other class of concern is separate from the strangeness of this new mechanism: the other concern is what then happens with those use cases. And I've certainly seen some examples like you've shown and heard some arguments that it's easy - if you have direct object access it is easy to get things wrong and that is certainly true. And it is certainly likely that there are alternative approaches than direct object access to another realm that would achieve the goals that I'm concerned about in ways that are harder to get wrong. I'm concerned that if this advances to stage 3 with this new form that has a restricted - like a subset of use cases met, that the sort of implied consensus that we also want to solve those other problems. Even if everyone may not have agreed with that like it would I would say implied is kind of gone. And that means that if I want to try and solve that use case, not only do I have to start from scratch with a new proposal, but I also have to come up with a mechanism that a lot of very experienced and intelligent people have not come up with, and admittedly weren't specifically trying to come up with, but it just kind of it's very strange to me that within stage 2 the shape of a solution would change so drastically such that it's almost a completely different proposal.
```